api/tests/integration/domain/services/user-reconciliation-service_test.js

```diff
@@ -0,0 +1,217 @@
+const { expect, sinon, domainBuilder } = require('../../../test-helper');
+const userReconciliationService = require('../../../../lib/domain/services/user-reconciliation-service');
+
+describe('Unit | Service | user-reconciliation-service', () => {
+
+  let students;
+  let user;
+
+  beforeEach(() => {
+    students = [
+      domainBuilder.buildStudent(),
+      domainBuilder.buildStudent(),
+    ];
+    user = {
+      firstName: 'Joe',
+      lastName: 'Poe',
+    };
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  context('When student list is not empty', () => {
+
+    context('When no student matched on names', () => {
+
+      it('should return an empty array if name is completely different', async () => {
+        // given
+        user.firstName = 'Sam';
+
+        students[0].firstName = 'Joe';
+        students[0].lastName = user.lastName;
+
+        // when
+        const result = await userReconciliationService.findMatchingCandidateIdForGivenUser(students, user);
+
+        // then
+        expect(result).to.equal(null);
```

Comment:
'should return an empty array...' , mais `null` est retourné

Reply:
en effet !